utils/resb/src/text/reader/parse_state.rs

```diff
@@ -66,7 +66,7 @@ impl<'a> ParseState<'a> {
 
     /// Takes the set of keys in order of discovery and replaces it with an
     /// empty set.
-    pub fn take_keys(self) -> IndexSet<Key<'a>> {
+    pub fn take_keys(&self) -> IndexSet<Key<'a>> {
```

Comment:
huh, is this a take? does it use interior mutability?

Reply:
Looks like it has a RefCell.

Maybe this fn should be renamed `into_keys()` since it seems like better code style to not use the class after taking the items out of the RefCell.

@leftmostcat thoughts?

Reply:
Also it's supposed to be called after all of parsing is done so that Rc should probably be unique, we could maybe call `mem::take(rc make_mut())`? deal with the error case somehow.

Reply:
Merging but hopeful for a better answer for the resb parser.